fpdf/image_parsing.py

```diff
@@ -127,24 +132,160 @@ def get_img_info(img, image_filter="AUTO", dims=None):
             "bpc": bpc,
             "f": image_filter,
-            "dp": f"/Predictor 15 /Colors {dpn} /Columns {w}",
+            "dp": dp,
         }
     )
 
     return info
 
+class temp_attr:
+    """
+    Copied verbatim from img2pdf
+
+    temporary change the attribute of an object using a context manager
+    """
+    def __init__(self, obj, field, value):
+        self.obj = obj
+        self.field = field
+        self.value = value
+
+    def __enter__(self):
+        self.exists = False
+        if hasattr(self.obj, self.field):
+            self.exists = True
+            self.old_value = getattr(self.obj, self.field)
+        print(f"setting {self.obj}.{self.field} = {self.value}")
+        setattr(self.obj, self.field, self.value)
+
+    def __exit__(self, exctype, excinst, exctb):
+        if self.exists:
+            setattr(self.obj, self.field, self.old_value)
+        else:
+            delattr(self.obj, self.field)
+
+def ccitt_payload_location_from_pil(img):
+    """
+    Copied verbatim from img2pdf
+    """
+    # If Pillow is passed an invalid compression argument it will ignore it;
+    # make sure the image actually got compressed.
+    if img.info["compression"] != "group4":
+        raise ValueError(
+            "Image not compressed with CCITT Group 4 but with: %s"
+            % img.info["compression"]
+        )
+
+    # Read the TIFF tags to find the offset(s) of the compressed data strips.
+    strip_offsets = img.tag_v2[TiffImagePlugin.STRIPOFFSETS]
+    strip_bytes = img.tag_v2[TiffImagePlugin.STRIPBYTECOUNTS]
+
+    # PIL always seems to create a single strip even for very large TIFFs when
+    # it saves images, so assume we only have to read a single strip.
+    # A test ~10 GPixel image was still encoded as a single strip. Just to be
+    # safe check throw an error if there is more than one offset.
+    if len(strip_offsets) != 1 or len(strip_bytes) != 1:
+        raise NotImplementedError(
+            "Transcoding multiple strips not supported by the PDF format"
+        )
+
+    (offset,), (length,) = strip_offsets, strip_bytes
+
+    #logger.debug("TIFF strip_offsets: %d" % offset)
+    #logger.debug("TIFF strip_bytes: %d" % length)
+
+    return offset, length
+
+
+def transcode_monochrome(imgdata):
+    """
+    Convert the open PIL.Image imgdata to compressed CCITT Group4 data.
+
+    Copied verbatim from img2pdf
+    """
+
+    #logger.debug("Converting monochrome to CCITT Group4")
+
+    # Convert the image to Group 4 in memory. If libtiff is not installed and
+    # Pillow is not compiled against it, .save() will raise an exception.
+    newimgio = BytesIO()
+
+    # we create a whole new PIL image or otherwise it might happen with some
+    # input images, that libtiff fails an assert and the whole process is
+    # killed by a SIGABRT:
+    #   https://gitlab.mister-muffin.de/josch/img2pdf/issues/46
+    im = Image.frombytes(imgdata.mode, imgdata.size, imgdata.tobytes())
```

Comment:
```suggestion
    img = Image.frombytes(imgdata.mode, imgdata.size, imgdata.tobytes())
```
This naming would be more consistent with the naming of other similar variables used in `fpdf2` code.

Reply:
there's another image named img but I've renamed this one img2